records/generator.go

```diff
@@ -375,37 +401,47 @@ func (rg *RecordGenerator) setFromLocal(host string, ns string) {
 	}
 }
 
-// insertRR inserts host to name's map
-// REFACTOR when storage is updated
-func (rg *RecordGenerator) insertRR(name string, host string, rtype string) {
-	logging.VeryVerbose.Println("[" + rtype + "]\t" + name + ": " + host)
-
+func (rg *RecordGenerator) exists(name string, host string, rtype string) bool {
 	if rtype == "A" {
 		if val, ok := rg.As[name]; ok {
 			// check if A record already exists
 			// identical tasks on same slave
 			for _, b := range val {
 				if b == host {
-					return
+					return true
 				}
 			}
-			rg.As[name] = append(val, host)
-		} else {
-			rg.As[name] = []string{host}
 		}
 	} else {
 		if val, ok := rg.SRVs[name]; ok {
 			// check if SRV record already exists
 			for _, b := range val {
 				if b == host {
-					return
+					return true
 				}
 			}
-			rg.SRVs[name] = append(val, host)
-		} else {
-			rg.SRVs[name] = []string{host}
 		}
 	}
+	return false
+}
+
+// insertRR adds a record to the appropriate record map for the given name/host pair,
+// but only if the pair is unique. returns true if added, false otherwise.
+// TODO(???): REFACTOR when storage is updated
+func (rg *RecordGenerator) insertRR(name string, host string, rtype string) bool {
+	logging.VeryVerbose.Println("[" + rtype + "]\t" + name + ": " + host)
+
+	if rg.exists(name, host, rtype) {
```

Comment:
i'm just maintaining existing functionality. i refactored so that I could use `exists` in a test. trying to keep the scope of this PR small.

that said, there's probably a lot of room for refactoring in the system. there's also lots of work planned for features and the bulk of this implementation is going to change at that point. probably not worth optimizing this now.